opensearch/src/main/java/org/opensearch/sql/opensearch/storage/OpenSearchIndex.java

```diff
@@ -84,14 +90,39 @@ public void create(Map<String, ExprType> schema) {
    *  Need to either handle field name conflicts
    *   or lazy evaluate when query engine pulls field type.
    */
+  /**
+   * Get simplified parsed mapping info. Unlike {@link #getFieldOpenSearchTypes()}
+   * it returns a flattened map.
+   * @return A map between field names and matching `ExprCoreType`s.
+   */
   @Override
   public Map<String, ExprType> getFieldTypes() {
+    if (cachedFieldOpenSearchTypes == null) {
+      cachedFieldOpenSearchTypes = new OpenSearchDescribeIndexRequest(client, indexName)
+          .getFieldTypes();
+    }
```

Comment:
duplicate with getFieldOpenSearchTypes()?

Reply:
Almost yes. `getFieldOpenSearchTypes` returns __full__ type info. It is used to create `OpenSearchExprValueFactory` - a class responsible to proper parsing values received from the cluster.
`getFieldTypes` used to create `TypeEnvironment` in `visitRelation`. `TypeEnvironment` keeps __simplified__ type info as `ExprCoreType`. It is used later for function resolution.